baseapp/baseapp.go

```diff
@@ -27,7 +28,7 @@ const (
 )
 
 var (
-	_ abci.Application = (*BaseApp)(nil)
+	_ tmabci.Application = (*BaseApp)(nil)
```

Comment:
there are some gotchas with this, since tendermint defines the types as tmproto we will need to do some type casting.